server/src/main/java/org/cloudfoundry/identity/uaa/login/TotpMfaEndpoint.java

```diff
@@ -179,18 +179,19 @@ private ModelAndView renderEnterCodePage(Model model, UaaPrincipal uaaPrincipal)
 
     private UaaPrincipal getSessionAuthPrincipal() throws UaaPrincipalIsNotInSession {
         UaaAuthentication uaaAuth = getUaaAuthentication();
-        if (uaaAuth != null) {
-            UaaPrincipal principal = uaaAuth.getPrincipal();
-            if (principal != null) {
-                return principal;
-            }
+        UaaPrincipal principal = uaaAuth.getPrincipal();
```

Comment:
Wouldn't scan mark it as null dereference because we are not checking uaaAuth.  I don't think it's smart enough to understand the new logic...

Reply:
Actually, it is pretty smart :) The issues fixed in this PR did go away when I ran the scan again using the PR branch: https://i.ci.build.ge.com/bjmjek1k/ci/job/UAA/job/Analysis/job/PR-266/2/